tools/src/main/java/org/apache/kafka/tools/ShareConsumerPerformance.java

```diff
@@ -0,0 +1,327 @@
+        if (!showDetailedStats)
+            System.out.printf("start.time, end.time, data.consumed.in.MB, MB.sec, data.consumed.in.nMsg, nMsg.sec%s%n", newFieldsInHeader);
+        else
+            System.out.printf("time, threadId, data.consumed.in.MB, MB.sec, data.consumed.in.nMsg, nMsg.sec%s%n", newFieldsInHeader);
+    }
+
+    private static void consume(KafkaShareConsumer<byte[], byte[]> shareConsumer,
+                                ShareConsumerPerfOptions options,
+                                AtomicLong totalMessagesRead,
+                                AtomicLong totalBytesRead,
+                                long bytesRead,
+                                long messagesRead,
+                                long lastBytesRead,
+                                long lastMessagesRead,
+                                AtomicLong joinTimeMsInSingleRound) {
+        long numMessages = options.numMessages();
+        long recordFetchTimeoutMs = options.recordFetchTimeoutMs();
+        long reportingIntervalMs = options.reportingIntervalMs();
+        boolean showDetailedStats = options.showDetailedStats();
+        SimpleDateFormat dateFormat = options.dateFormat();
+        shareConsumer.subscribe(options.topic());
+
+        // now start the benchmark
+        long currentTimeMs = System.currentTimeMillis();
+        long lastReportTimeMs = currentTimeMs;
+        long lastConsumedTimeMs = currentTimeMs;
+
+        while (messagesRead < numMessages && currentTimeMs - lastConsumedTimeMs <= recordFetchTimeoutMs) {
+            ConsumerRecords<byte[], byte[]> records = shareConsumer.poll(Duration.ofMillis(100));
+            currentTimeMs = System.currentTimeMillis();
+            if (!records.isEmpty())
+                lastConsumedTimeMs = currentTimeMs;
+            for (ConsumerRecord<byte[], byte[]> record : records) {
+                messagesRead += 1;
+                if (record.key() != null)
+                    bytesRead += record.key().length;
+                if (record.value() != null)
+                    bytesRead += record.value().length;
+                if (currentTimeMs - lastReportTimeMs >= reportingIntervalMs) {
+                    if (showDetailedStats)
+                        printConsumerProgress(0, bytesRead, lastBytesRead, messagesRead, lastMessagesRead,
+                            lastReportTimeMs, currentTimeMs, dateFormat, joinTimeMsInSingleRound.get());
+                    joinTimeMsInSingleRound = new AtomicLong(0);
+                    lastReportTimeMs = currentTimeMs;
+                    lastMessagesRead = messagesRead;
+                    lastBytesRead = bytesRead;
+                }
+            }
+        }
+
+        if (messagesRead < numMessages)
+            System.out.printf("WARNING: Exiting before consuming the expected number of messages: timeout (%d ms) exceeded. " +
+                "You can use the --timeout option to increase the timeout.%n", recordFetchTimeoutMs);
```

Comment:
I don't think there is a timeout option.

Reply:
we do have a --timeout option passed as command line argument mentioned [here](https://github.com/confluentinc/kafka/pull/1297/files#diff-09fe24f771196ae970f441d47acc2be8a1ccc484b1a4f78f31bef8c2c06ac663R234) , otherwise the timeout defaults to 10 seconds.

Reply:
Ah, I did check for it but missed it. Sorry.